src/Processors/Merges/Algorithms/ReplacingSortedAlgorithm.cpp

```diff
@@ -93,6 +93,10 @@ IMergingAlgorithm::Status ReplacingSortedAlgorithm::merge()
         }
     }
 
+    /// If have enough rows, return block, because it prohibited to overflow requested number of rows.
+    if (merged_data.hasEnoughRows())
+        return Status(merged_data.pull());
+
```

Comment:
This fixes `Incomplete granules are not allowed while blocks are granules size`.